### PR TITLE
server: separate store info and add uptime

### PR DESCRIPTION
**Pull request by huachaohuang**
This PR separates the `storeInfo` from the cache to cleanup code of the cache.
A `StartTS` is added to the `StoreStatus` to indicate when the store is started and how long the store has been up.

Close Issue: #339

### CODE REVIEW

File: pkg/timeutil/duration.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+// Duration is a wrapper of time.Duration for TOML and JSON.
+type Duration struct {
```
Comment (siddontang) [inline pkg/timeutil/duration.go]: any test?

File: server/store.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+func (s *StoreStatus) update(stats *pdpb.StoreStats, totalRegionCount, leaderRegionCount int) {
+	s.StoreStats = stats
+	s.LastHeartbeatTS = time.Now()
+	s.TotalRegionCount = totalRegionCount
+	s.LeaderRegionCount = leaderRegionCount
+}
+
+// GetUptime returns the uptime of the store.
+func (s *StoreStatus) GetUptime() time.Duration {
+	uptime := s.LastHeartbeatTS.Sub(s.StartTS)
```
Comment (siddontang) [inline server/store.go]: Any where to change the StartTS? 
If the store is down for a long time, what's the up time?

Comment (huachaohuang) [inline server/store.go]: The `StartTS` will be set when the `StoreStatus` is created, that is, when a TiKV restarts and sends a `PutStore` request to PD.
If the store is down, `LastHeartbeatTS` will not be updated, so the uptime remains the same, until the store recovers or restarts.

Comment (siddontang) [inline server/store.go]: Got it.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @overvenus

File: server/api/store.go [hunk 3]
```diff
@@ -19,13 +19,31 @@ import (
 
 	"github.com/gorilla/mux"
 	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/pkg/timeutil"
 	"github.com/pingcap/pd/server"
 	"github.com/unrolled/render"
 )
 
+type storeStatus struct {
+	*server.StoreStatus
```
Comment (overvenus) [inline server/api/store.go]: Missing `json:"StoreStatus"` or we don't need to write it explicit?

Comment (huachaohuang) [inline server/api/store.go]: I don't write it in purpose, to make the JSON output clean.

Comment (overvenus) [inline server/api/store.go]: LGTM

**Pull request merged**